olive/passes/onnx/optimum_merging.py

```diff
@@ -75,11 +77,23 @@ def new_byte_size_func(_):
         olive_model = model_proto_to_olive_model(merged_model, output_model_path, config)
 
         # Doing a dry run of ORT allows us to remove the initializers that were orphaned by the merging step
-        sess_options = onnxruntime.SessionOptions()
-        sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_BASIC
+        sess_options = ort.SessionOptions()
+        sess_options.graph_optimization_level = ort.GraphOptimizationLevel.ORT_ENABLE_BASIC
         sess_options.optimized_model_filepath = output_model_path
 
         execution_provider = self.accelerator_spec.execution_provider
-        onnxruntime.InferenceSession(output_model_path, sess_options, providers=[execution_provider])
+        sess_kwargs = {}
+        if is_winml_installation():
+            ort_device_type = get_ort_hardware_device_type(self.accelerator_spec.accelerator_type)
+            sess_options.add_provider_for_devices(
+                [
+                    ep_device for ep_device in ort.get_ep_devices()
+                    if ep_device.device.type == ort_device_type and ep_device.ep_name == execution_provider
+                ],
+                {},
+            )
+        else:
+            sess_kwargs = {"providers": [execution_provider]}
+        ort.InferenceSession(output_model_path, sess_options, **sess_kwargs)
```

Comment:
There are 3 similar part of this code in this PR. Can they be abstracted?

Reply:
Consolidated.